Animals/Animals/Classes/Animal.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Animals
+{
+    public enum AnimalTypes { Bird, Fish, Insect, Reptile, Mammal }
+    public abstract class Animal
+    {
+        public string Name { get; set; }
+
+        public Animal(string name)
+        {
+            Name = name;
+        }
+
+        public virtual void Fly()
```

Comment:
citeste enuntul!
la tine orice animal va zbura, merge si inota si se va tarai.

Reply:
M-a incurcat "no matter the category animal are flying, walking, swim or crawl".
de aceea am implementat la nivel de animal metodele astea, urmand ca, la fiecare in parte sa fac override atunci cand fac animalul respectiv sa faca metoda.
Adica mai pe scurt, il pun pe delfin sa mearga si el zice ca nu poate.
Ma gandesc atunci la alta abordare.